### DISCUSSION TITLE
How does Four Keys detect production deployments with Gitlab?

### DISCUSSION THREAD

**Discussion by hameno** [GoogleCloudPlatform/fourkeys#392] [Q&A]
We're trying to implement Four Keys in our team. We are using Gitlab and use a branching strategy that more or less follows Gitflow.
This means we have a develop and a master branch. Therefore not every deployment is a production deployment.
So my question is: Can Four Keys handle this? 
Follow up: If we would be switching to a one-main branch setup, we would still have multiple deployments per commit (dev, stage and prod) - how would this work? I have seen no indication/configuration that matches the deployment tier/environment name

**Comment by colinfrei** [comment]
I'm looking into this as well, and I also feel that the default Four Keys queries aren't exact enough.

Looking at this query: https://github.com/dora-team/fourkeys/blob/main/queries/deployments.sql

If you're using Deployments you can probably add a condition to identify the deployment environment (around line 32)
We're not using Deployments in Gitlab, so I'm looking into checking Jobs, and there looking for the specific job name of our deployment job. Not ideal either, but that's my current attempt :) My code looks like this:
`      OR (source LIKE "gitlab%" AND event_type = "build" AND JSON_EXTRACT_SCALAR(metadata, '$.build_status') = "success" AND JSON_EXTRACT_SCALAR(metadata, '$.build_name') = "deploy_to_gcp_prod")`

Not sure how it'd work with pipelines.
